pkg/controller/direct/compute/forwardingrule_controller.go

```diff
@@ -391,49 +444,33 @@ func (a *forwardingRuleAdapter) Update(ctx context.Context, updateOp *directbase
 	var err error
 	op := &gcp.Operation{}
 	updated := &computepb.ForwardingRule{}
-	// TODO(yuhou): Checked the realGCP logs, setLabels request is being sent even when there are no updates to labels.
-	// That might because of the generated labelsFingerPrint?
-	if a.id.location == "global" {
-		setLabelsReq := &computepb.SetLabelsGlobalForwardingRuleRequest{
-			Resource:                       a.id.forwardingRule,
-			GlobalSetLabelsRequestResource: &computepb.GlobalSetLabelsRequest{LabelFingerprint: a.actual.LabelFingerprint, Labels: forwardingRule.Labels},
-			Project:                        a.id.project,
-		}
-		op, err = a.globalForwardingRulesClient.SetLabels(ctx, setLabelsReq)
-	} else {
-		setLabelsReq := &computepb.SetLabelsForwardingRuleRequest{
-			Resource:                       a.id.forwardingRule,
-			RegionSetLabelsRequestResource: &computepb.RegionSetLabelsRequest{LabelFingerprint: a.actual.LabelFingerprint, Labels: forwardingRule.Labels},
-			Project:                        a.id.project,
-			Region:                         a.id.location,
+	if !reflect.DeepEqual(forwardingRule.Labels, a.actual.Labels) {
+		if a.id.location == "global" {
```

Comment:
Nit: looks like we could refactor out a `a.setLabels` helper to get rid of the duplicated setLabels code ?